### PR TITLE
refactor: Object.assign -> object spread

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Refactors Object.assign use to Object spread.

## Context:

Better type checking.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by JamieMagee — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4176,7 +4176,7 @@ fast-base64-decode@^1.0.0:
   resolved "https://registry.yarnpkg.com/fast-base64-decode/-/fast-base64-decode-1.0.0.tgz#b434a0dd7d92b12b43f26819300d2dafb83ee418"
   integrity sha512-qwaScUgUGBYeDNRnbc/KyllVU88Jk1pRHPStuF/lO7B0/RTRLj7U0lkdTAutlBblY08rwZDff6tNU9cjv6j//Q==
 
-fast-deep-equal@3.1.3, fast-deep-equal@^3.1.1:
+fast-deep-equal@^3.1.1:
```
Comment (viceice) [inline yarn.lock:4179]: Why this change?

**Review by viceice — CHANGES_REQUESTED** [review]
Needs fixes

**Review by viceice — APPROVED** [review]
